api/v1/api.go

```diff
@@ -210,3 +209,71 @@ func (api *API) respondError(w http.ResponseWriter, apiErr apiError, data interf
 		level.Error(api.logger).Log("msg", "failed to write data to connection", "err", err)
 	}
 }
+
+type metricJson map[string]interface{}
```

Comment:
Should be `metricJSON` according to general Go style.

But since this is not directly JSON, perhaps call it `encodableMetric` or something?